pkg/kubelet/deviceplugin/endpoint.go

```diff
@@ -74,7 +74,13 @@ func newEndpoint(socketPath, resourceName string, callback MonitorCallback) (*en
 func (e *endpoint) getDevices() []*pluginapi.Device {
```

Comment:
I am trying to see the value of using *pluginapi.Device instead of pluginapi.Device here and also in endpoint.devices. AFAICT, we are still doing allocation and copy almost everywhere when we need to access devices in this file. Would code be simplified if we just use pluginapi.Device? I think we can at least get rid of cloneDevices().

Reply:
This seems pretty reasonable

Reply:
agree